src/Parser.php

```diff
@@ -0,0 +1,288 @@
+<?php
+namespace DemandwareXml;
+
+use \SimpleXMLElement;
+use \XMLReader;
+
+/**
+ * Parses a Demandware XML file into the six main data structures expected, and returns arrays for ease of working with
+ *
+ * @package DemandwareXml
+ * @todo: not currently needed, but iterator (or better, generators? http://evertpot.com/switching-to-generators/) may be more memory efficient than arrays for large files
+ */
+class Parser
+{
+    private $assignments = [];
+    private $bundles     = [];
+    private $categories  = [];
+    private $products    = [];
+    private $sets        = [];
+    private $variations  = [];
+
+    /**
+     * Create a new parser for the specified path, which will be validated against the XSD before parsing
+     *
+     * @param $path
+     * @throws Exception
+     * @throws XmlException
+     */
+    public function __construct($path)
+    {
+        $reader = new XMLReader;
+
+        if (! $reader->open($path)) {
+            throw new XmlException('Error opening ' . $path);
+        }
+
+        Xml::validate($path);
+
+        $this->parse($reader);
+    }
+
+    /**
+     * Return an array containing product ids as keys, and an associative array of category ids mapped to whether primary for the values
+     *
+     * @return array
+     */
+    public function assignments()
```

Comment:
I think these should probably be `getAssignments()`, `getCategories()` etc. so we have consistent getters and setters.